Content.Server/GameObjects/Components/Power/PoweredLightComponent.cs

```diff
@@ -1,49 +1,171 @@
 using System;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces.GameObjects;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.Components.Inventory;
 using SS14.Server.GameObjects;
+using SS14.Server.GameObjects.Components.Container;
 using SS14.Server.GameObjects.EntitySystems;
 using SS14.Shared.Audio;
 using SS14.Shared.Enums;
 using SS14.Shared.GameObjects;
 using SS14.Shared.Interfaces.GameObjects;
 using SS14.Shared.Interfaces.Timing;
 using SS14.Shared.IoC;
+using SS14.Shared.Log;
+using SS14.Shared.Map;
+using SS14.Shared.Serialization;
+using SS14.Shared.ViewVariables;
 
 namespace Content.Server.GameObjects.Components.Power
 {
-    public class PoweredLightComponent : Component
+    public class PoweredLightComponent : Component, IAttackHand, IAttackby
     {
         public override string Name => "PoweredLight";
 
         private static readonly TimeSpan _thunkDelay = TimeSpan.FromSeconds(2);
 
         private TimeSpan _lastThunk;
 
-        public override void Initialize()
+        private LightBulbType BulbType = LightBulbType.Tube;
+
+        private float Load = 40;
+
+        [ViewVariables] private ContainerSlot LightBulbContainer;
+
+        [ViewVariables]
+        private LightBulbComponent LightBulb
         {
-            base.Initialize();
+            get
+            {
+                if (LightBulbContainer.ContainedEntity == null) return null;
+
+                LightBulbContainer.ContainedEntity.TryGetComponent(out LightBulbComponent bulb);
+
+                return bulb;
+            }
+        }
+
+        bool IAttackby.Attackby(IEntity user, IEntity attackwith)
+        {
+            if (!attackwith.HasComponent<LightBulbComponent>()) return false;
+
+            if (LightBulb != null) return false;
+
+            user.GetComponent<IHandsComponent>().Drop(attackwith, LightBulbContainer);
+
+            var inserted = LightBulbContainer.Insert(attackwith);
```

Comment:
Pretty sure this insert already gets done by the drop call, doesn't it?

Reply:
Yes, HandheldLightComponent does that too, so I thought it was necessary.